protocol-jackson/README.md

```diff
@@ -0,0 +1,203 @@
+# Jackson bindings for Zeebe protocol records
+
+The goal of this module is to provide a basic implementation of the [Zeebe protocol](/protocol)
+`Record` interface and its possible values, e.g. `VariableRecordValue`, `IncidentRecordValue`, etc.
+
+This implementation can be used to (de)serialize these objects using Jackson (and therefore any
+format supported by Jackson). Use cases here are mostly for exporters and Java applications that are
+built around exporters, such as Operate, Tasklist, and Optimize.
+
+Additionally, the module provides builders for each of these values, allowing them to be used as
+standalone `Record` implementations. The main use case here is for testing, where it may be useful
+to specific records.
+
+## How it works
+
+The module provides a set of abstract types and uses [Immutables](https://immutables.github.io) to
+generate the actual implementation. To reduce API surface, the generated immutable classes are
+hidden, and instead only the generated builders and the abstract stubs are exposed.
+
+> NOTE: hiding the implementation classes is achieved via configuration in the `ZeebeStyle`
+> annotation. See there for more comments on which options control what.
+
+Serialization to and from the hidden generated types still work, as users can use the abstract
+types, e.g. `AbstractVariableRecordValue`, as the type reference in Jackson. These abstract types
+are annotated to tell Jackson what concrete type should be used for deserialization.
+
+Serializing an immutable record would just be like serializing any POJO with Jackson:
+
+```java
+// assume a pre-defined Record<?> record object
+final ObjectMapper mapper=new ObjectMapper();
+final String myJson=mapper.writeValueAsString(record);
+```
```

Comment:
🙃  Why are you consistently not using a space around the equals char?

Reply:
That's the auto formatter :smile: